runtime-sdk/src/context.rs

```diff
@@ -318,6 +334,10 @@ impl<'a, R: runtime::Runtime, S: storage::Store> Context for RuntimeBatchContext
         };
         f(sim_ctx)
     }
+
+    fn check_invariants(&mut self) -> bool {
```

Comment:
Would it be nicer if this returned a `Result<(), Error>`. The invariant checks can then return more descriptive errors.

Reply:
Yeah, that was my intention initially as well... but, how do you properly merge the Errors of all InvariantHandlers for all modules? (see `impl InvariantHandler for Tuple` in `runtime-sdk/src/module.rs`)
If I had used the same trick as was used a few lines above for the Genesis parameters in the migration handler, the error codes would be mixed up (e.g. two modules can define error code 2 to mean different errors).
I suppose we could have errors without codes for this or come up with some other scheme?

Reply:
OK, I'll try it out.